### ISSUE TITLE
[LOW] [P2P] [$500] Expense - Mentioning payment sender in IOU thread says user isn't a member of this room

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#35166]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.31.1
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @shawnborton 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1706185536516179

## Action Performed:
1. User A sends a payment to User B. Then User B @mentions user A in that same IOU thread.

## Expected Result:
Whisper saying user is not a member of this room should not be shown.

## Actual Result:
Whisper saying user is not a member of this room is shown.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

![CleanShot 2024-01-25 at 07 23 56@2x](https://github.com/Expensify/App/assets/93399543/50ec11b7-7958-4686-bbf8-b91f39b70d22)

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~016219e2ce558b977a</li>
        <li>Upwork Job ID: 1750564810849771520</li>
        <li>Last Price Increase: 2024-02-01</li>
    </ul>
</details>

**Comment by kbecciv** [comment]
We think that this bug might be related to #vip-vsb
CC @quinthar

**Comment by FitseTLT** [comment]
I think this should be fixed on BE because it is nice to have the mention although it is not a room but it shouldn't try to join the user to the report if it is not a room that the comment is being made.

**Comment by jliexpensify** [comment]
Posted here - https://expensify.slack.com/archives/C066HJM2CAZ/p1706488633882819

**Comment by jliexpensify** [comment]
@sobitneupane could you confirm this [comment?](https://github.com/Expensify/App/issues/35166#issuecomment-1910638262) If so, we can assign this as `Internal`, thanks!

**Comment by jliexpensify** [comment]
@sobitneupane [bump on my question!](https://github.com/Expensify/App/issues/35166#issuecomment-1913996775)

**Comment by sobitneupane** [comment]
I couldn't find the frontend logic responsible for triggering user invitations when mentioned. It's likely handled on the backend, so we can label this as an internal task.

**Comment by jliexpensify** [comment]
Can't auto-assign an Engineer so posted here: https://expensify.slack.com/archives/C05RECHFBEW/p1706837066213689

**Comment by jliexpensify** [comment]
This is not overdue!

**Comment by jliexpensify** [comment]
Waiting on an Internal Engineer to pick this up cc @arielgreen

**Comment by jliexpensify** [comment]
Added `Hot Pick` label

**Comment by jliexpensify** [comment]
Changed to `Weekly` as we wait for an Internal Engineer to pick this up

**Comment by jliexpensify** [comment]
Still waiting on an Internal Engineer

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by jliexpensify** [comment]
Cool, going to close this!
